### PR TITLE
Perf fix for ProcessHelper.TerminateProcess

### PR DESCRIPTION
**Pull request by harshjain2**
Modified the TerminateProcess API to pass Process as argument instead of Process.Id.
This gives a perf benefit of around 100 milliseconds as we used to get Process object using ProcessId which we already have.

### CODE REVIEW

File: src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs [hunk 1]
```diff
@@ -127,10 +127,9 @@ public void SetExitCallback(int processId, Action callbackAction)
         }
 
         /// <inheritdoc/>
-        public void TerminateProcess(int processId)
+        public void TerminateProcess(object process)
         {
-            var process = Process.GetProcessById(processId);
-            process.Kill();
+            ((Process)process).Kill();
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs]: please add a null check after Typecasting, if something else is passed other than process object

**Review by mayankbansal018 — APPROVED** [review]

File: test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Hosting/DefaultTestHostManagerTests.cs [hunk 2]
```diff
@@ -303,19 +303,19 @@ public void LaunchTestHostShouldUseCustomHostIfSet()
             await this.testableTestHostManager.LaunchTestHostAsync(this.GetDefaultStartInfo(), CancellationToken.None);
             await this.testableTestHostManager.CleanTestHostAsync(CancellationToken.None);
 
-            this.mockProcessHelper.Verify(ph => ph.TerminateProcess(Process.GetCurrentProcess().Id), Times.Once);
+            this.mockProcessHelper.Verify(ph => ph.TerminateProcess(It.IsAny<Process>()), Times.Once);
```
Comment (codito) [inline test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Hosting/DefaultTestHostManagerTests.cs]: How do we validate that it is the same process object that's provided?

File: src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs (hunk 1 above)
Comment (harshjain2) [inline src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs]: done.

**Comment by harshjain2** [comment]
@dotnet-bot test Windows_NT / Release Build

**Pull request merged**